pkg/helmreconciler/manifest_rendering.go

```diff
@@ -0,0 +1,119 @@
+package helmreconciler
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	binversion "istio.io/operator/version"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/helm/pkg/manifest"
+	"k8s.io/helm/pkg/releaseutil"
+
+	"istio.io/operator/pkg/component/controlplane"
+	operatormanifest "istio.io/operator/pkg/manifest"
+	"istio.io/operator/pkg/name"
+	"istio.io/operator/pkg/translate"
+)
+
+var (
+	kindRegex = regexp.MustCompile("kind:(.*)\n")
+	componentToChartName = map[name.ComponentName]string{
+		name.IstioBaseComponentName:       "istio",
+		name.PilotComponentName:           "istio/charts/pilot",
+		name.GalleyComponentName:          "istio/charts/galley",
+		name.SidecarInjectorComponentName: "istio/charts/sidecarInjectorWebhook",
+		name.PolicyComponentName:          "istio/charts/mixer-policy",
+		name.TelemetryComponentName:       "istio/charts/mixer-telemetry",
+		name.CitadelComponentName:         "istio/charts/security-citadel",
+		name.CertManagerComponentName:     "istio/charts/security-certmanager",
+		name.NodeAgentComponentName:       "istio/charts/security-nodeagent",
+		name.IngressComponentName:         "istio/charts/gateways-istio-engress",
+		name.EgressComponentName:          "istio/charts/gateways-istio-egress",
+	}
+)
+
+func (rc *ISCPReconciler) renderManifests(input RenderingInputV2) (ChartManifestsMap, error) {
+	cr, err := ioutil.ReadFile(input.GetCRPath())
+	if err != nil {
+		return nil, fmt.Errorf("could not read cr file %s: %s", input.GetCRPath(), err)
+	}
+	icps, _, err := operatormanifest.ParseK8SYAMLToIstioControlPlaneSpec(string(cr))
+	if err != nil {
+		return nil, fmt.Errorf("could not IstioControlPlane CR YAML:\n%s", cr)
+	}
+	t, err := translate.NewTranslator(binversion.OperatorBinaryVersion.MinorVersion)
+	if err != nil {
+		return nil, err
```

Comment:
can we give an error msg here too?

Reply:
errors should be returned in the stack rather than logged per function.  I think the assumption here is that the result of translate.NewTranlater is properly formatted.